### PR TITLE
Yolo loss darknet

### PR DESCRIPTION
**Pull request by heavengate**

### CODE REVIEW

File: paddle/fluid/operators/yolov3_loss_op.h [hunk 1]
```diff
@@ -25,304 +26,222 @@ template <typename T, int MajorType = Eigen::RowMajor,
...
-using Array5 = Eigen::DSizes<int64_t, 5>;
+template <typename T>
+static inline bool LessEqualZero(T x) {
+  return x < 1e-6;
+}
 
 template <typename T>
-static inline bool isZero(T x) {
-  return fabs(x) < 1e-6;
+static T SCE(T x, T label) {
```
Comment (qingqing01) [inline paddle/fluid/operators/yolov3_loss_op.h]: What is the meaning of `SCE` ?

File: python/paddle/fluid/layers/detection.py [hunk 2]
```diff
@@ -413,13 +413,10 @@ def yolov3_loss(x,
...
                 anchors,
+                anchor_mask,
                 class_num,
                 ignore_thresh,
-                loss_weight_xy=None,
-                loss_weight_wh=None,
-                loss_weight_conf_target=None,
-                loss_weight_conf_notarget=None,
-                loss_weight_class=None,
+                downsample,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: downsample -> downsample_ratio

**Review by qingqing01 — COMMENTED** [review]
Put yolov3_loss_op into `paddle/fluid/operators/detection` directory

File: python/paddle/fluid/layers/detection.py (hunk 2 above)
Comment (heavengate) [reply python/paddle/fluid/layers/detection.py]: Done

File: paddle/fluid/operators/yolov3_loss_op.h (hunk 1 above)
Comment (heavengate) [reply paddle/fluid/operators/yolov3_loss_op.h]: change to SigmoidCrossEntropy

**Review by heavengate — COMMENTED** [review]
Done

**Review by qingqing01 — APPROVED** [review]
LGTM.

**Review by XiaoguangHu01 — APPROVED** [review]
同意合入

**Comment by panyx0718** [comment]
note: this api change is not backward compatible

**Pull request merged**